src/soilbiogeochem/SoilBiogeochemCarbonFluxType.F90

```diff
@@ -229,10 +229,12 @@ subroutine InitHistory(this, bounds, carbon_type)
              avgflag='A', long_name='total heterotrophic respiration', &
              ptr_col=this%hr_col)
 
-        this%michr_col(begc:endc) = spval
-        call hist_addfld1d (fname='MICC_HR', units='gC/m^2/s', &
+        if (decomp_method == mimics_decomp) then
+           this%michr_col(begc:endc) = spval
+           call hist_addfld1d (fname='MICC_HR', units='gC/m^2/s', &
              avgflag='A', long_name='microbial C heterotrophic respiration: donor-pool based, so expect zero with MIMICS', &
              ptr_col=this%michr_col, default='inactive')
+        end if
```

Comment:
Surrounded the "MIC" variables with mimics if-statements to prevent from sending to history when not mimics.

The MICC_HR fields are already inactive, so nothing will appear different in history for these.

The TOTMIC* fields are active, so they will stop appearing in history when not mimics.

There are a few more fields that we could stop sending to history when not running mimics, but they require additional if-statements as we loop through `ndecomp_cascade_transitions` or `ndecomp_pools`. Shall I go on to do that?

Reply:
> There are a few more fields that we could stop sending to history when not running mimics, but they require additional if-statements as we loop through `ndecomp_cascade_transitions` or `ndecomp_pools`. Shall I go on to do that?

In `SoilBiogeochemCarbonFluxType.F90` all such fields are already "inactive" so probably not worth adding if-statements to those.

In `SoilBiogeochemCarbonFluxType.F90` T_SCALAR is Century-specific, so I'm putting an if-statement around that one.

Reply:
So for these cases where fields are inactive, but you are running with the other decomp_type, will the fields in question be filled with zero's? Or will they be nan's or not allocated? If so that would allow a situation where you ask for them and the model blows up. That would be a reason to add if statements around hist_addfld calls. It's probably OK if they just get set to zero, but if it blows up because they aren't allocated or they are nan's that could be a problem.

Reply:
I ran a simulation with CENTURY and one with MIMICS and looked at MICC_HR and found that it looks the same. Zeros over land and 1e36 over ocean.

Also, I looked at the rest of the variables that say "inactive" and they are valid variables for both CENTURY and MIMICS. I think the test-suite will raise red flags if I am wrong. I will run the test suite on this tomorrow.

Reply:
> I think the test-suite will raise red flags if I am wrong. I will run the test suite on this tomorrow.

...if I make all of them active for the test.

Reply:
OK, sounds good. We really need a flag to turn on all the history variables, so that we can check for validity. But, that's outside what you are doing here.